modules/AOW_Conditions/AOW_Condition.php

```diff
@@ -122,7 +124,16 @@ public function save_lines($post_data, $parent, $key = '')
                             }
                         } else {
                             if ($field_name === 'value' && $post_data[$key . 'value_type'][$i] === 'Value') {
-                                $post_data[$key . $field_name][$i] = fixUpFormatting($_REQUEST['flow_module'], $condition->field, $post_data[$key . $field_name][$i]);
+                                $condition_module = $_REQUEST['flow_module'];
+                                if ($condition_relation !== $condition_module){
+                                    $flow_bean = new $GLOBALS['beanList'][$_REQUEST['flow_module']];
+                                    $flow_bean->load_relationship($condition_relation);
+                                    $rel_module = $flow_bean->$condition_relation->relationship->lhsLinkDef['module'];
+                                    if (!empty($rel_module) && $_REQUEST['flow_module'] !== $rel_modules) {
+                                        $condition_module = $rel_module;
+                                    }
+                                }
```

Comment:
Hi @re8260,

Thank you for your PR.

@re8260, @serhiisamko091184 

To avoid warnings and notices and also for security reasons it is probably best to add some more checks. the following is my suggestion:

```
                                $condition_module = $_REQUEST['flow_module'] ?? '';
                                $condition_module_bean = $GLOBALS['beanList'][$condition_module] ?? '';
                                 
                                if ($condition_relation !== $condition_module && $condition_module_bean !== ''){
                                    $flow_bean = new $condition_module_bean();
                                    $flow_bean->load_relationship($condition_relation);
                                    $rel_module = $flow_bean->$condition_relation->relationship->lhsLinkDef['module'];
                                    if (!empty($rel_module) &&  $condition_module !== $rel_modules) {
                                        $condition_module = $rel_module;
                                    }
                                }

```